### PR TITLE
Loki: Move explain section to builder mode

### LINKED ISSUE #52980: Loki/Prometheus: Integrate explain mode into builder mode
### Discussed in https://github.com/grafana/grafana/discussions/52099

<div type='discussions-op-text'>

<sup>Originally posted by **ivanahuckova** July 12, 2022</sup>
Currently Explain mode is separate, but having it directly in the Builder mode could be better for users. </div>

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This PR is based on https://github.com/grafana/grafana/pull/50437/commits/d924e5a2ad25074d53eef4c59452ed4f1d90e796. IT moves Explain section to Code and Builder mode in Loki and Prometheus. 

https://user-images.githubusercontent.com/30407135/181256536-fd4315e9-2438-46e1-b30a-c21fcd61d7b1.mov

As a part of this PR, we are removing Explain query builder mode type as it is not needed anymore.

Known issues that we could tackle in separate PRs to make this one smaller: 
1. Shared: The transition (blue highlight around operation) isn't the best. Maybe we'll need to use react transition api
2. Loki, Prometheus: Documentation needs to be updated
3. Possibly add more tests for Explain section in builder and code editor

Fixes: [#52980](https://github.com/grafana/grafana/issues/52980)

### CODE REVIEW

**Comment by torkelo** [comment]
Nice! 🙌

File: public/app/plugins/datasource/loki/querybuilder/state.ts [hunk 1]
```diff
@@ -80,3 +80,28 @@ export function useRawQuery(): [boolean, (val: boolean) => void] {
 
   return [rawQuery, setter];
 }
+
+const queryEditorExplainLocalStorageKey = 'LokiQueryEditorExplainDefault';
+
+function getExplainVisibility(): boolean {
+  const val = store.get(queryEditorExplainLocalStorageKey);
+  return val === undefined ? true : Boolean(parseInt(val, 10));
```
Comment (itsmylife) [inline public/app/plugins/datasource/loki/querybuilder/state.ts:88]: Do we want to show explain mode by default?

Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/querybuilder/state.ts:88]: Good question, maybe we can switch it to false. Yeah, I think default to false is fine.

Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/querybuilder/state.ts:88]: Do you want to do it as a part of moving hook to to shared file?

Comment (itsmylife) [reply public/app/plugins/datasource/loki/querybuilder/state.ts:88]: Yeah I will do it in the hook ;)

**Review by itsmylife — APPROVED** [review]
LGTM 🚀

**Pull request merged**